app/component/itinerary/NaviContainer.js

```diff
@@ -96,6 +97,10 @@ function NaviContainer(
           return { ...l };
         });
         setRealTimeLegs(rtLegs);
+        setLatestNavigatorItinerary({
+          ...itinerary,
+          legs: rtLegs,
+        });
       });
```

Comment:
I wonder if it is necessary to re-save the itinerary all the time? Page reload might briefly show initial itinerary before it gets updated, but I am not sure if this is a problem. what is your opinion? Maybe the realtime query should be made when the component mounts.

Reply:
Refactoring the component to send the first query on mount sounds like a better solution. I'll refactor.